File: Easy/RomanToInteger.py
# https://leetcode.com/problems/roman-to-integer/

class Solution:
    def roman_to_integer(self, s: str) -> int:
        romanValues = {'I': 1, 'V': 5, 'X': 10,
                       'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        result = 0
        i = 0
        while (i < len(s)-1):
            current_value = romanValues[s[i]]
            next_value = romanValues[s[i+1]]
            if current_value < next_value:
                result += next_value - current_value
                i += 2
            else:
                result += current_value
                i += 1
            
        if (i == len(s)-1):
            result += romanValues[s[i]]
        return result
            


s = input().upper()
a = Solution()
print(a.roman_to_integer(s))

# MCMXCIV :: Input
# 1994 :: Output

File: Easy/RomanToInteger.java
package Easy;

// https://leetcode.com/problems/roman-to-integer/
import java.util.HashMap;
import java.util.Scanner;

public class RomanToInteger {
    public static int romanToInt(String s ){
        int result = 0;
        
        HashMap<String, Integer> romanValue = new HashMap<>();
        char[] str = s.toCharArray();
        romanValue.put("I", 1); 
        romanValue.put("V", 5);
        romanValue.put("X", 10);
        romanValue.put("L", 50);
        romanValue.put("C", 100);
        romanValue.put("D", 500);
        romanValue.put("M", 1000);
        int i = 0;
        while(i<str.length-1){
            System.out.println("i : " + i);
            int value = romanValue.get(String.valueOf(str[i]));
            System.out.println("Value: " + value);
            int nextVAlue = romanValue.get(String.valueOf(str[i+1]));
            System.out.println("nextValue : " + nextVAlue);

            if (value < nextVAlue){
                result += nextVAlue-value;
                i += 2;
            }
            else{
                result += value;
                i++;
            }
            
            System.out.println("Result : " + result);

            
        }
        if (i == str.length-1){
            result += romanValue.get(String.valueOf(str[i]));
        }
        return result;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine().toUpperCase();
        sc.close();
        System.out.println(romanToInt(str));

    }
}
// MCMXCIV :: Input
// 1994 :: Output